contents/dotfiles/Dotfiles.md

```diff
@@ -0,0 +1,233 @@
+```
+Host remoteA
+    Hostname remoteAserver.com
+    User john
+    IdentityFile ~/.ssh/id_rsa
+```
+
+With the configurations above, John is now able to simply just run the following to connect to `remoteA`:
+```bash
+ssh remoteA
+```
+This evidently saves time and effort needed to connect to a server. Aside from just connecting to servers, `ssh` can also be used for other purposes, all of which are easily [configurable](https://www.ssh.com/ssh/config) using dotfiles. These include [SSH proxies](https://www.cyberciti.biz/faq/linux-unix-ssh-proxycommand-passing-through-one-host-gateway-server/), [launching GUI applications remotely](https://www.nics.tennessee.edu/x11_forwarding) and [SSH forwarding](https://nerderati.com/2011/03/17/simplify-your-life-with-an-ssh-config-file/).
+
+<box type="warning">
+Please ensure that your RSA private keys such as `~/.ssh/id_rsa` are kept safe at all times.
+</box>
+
+#### Benefit 2: Personalised Shortcuts and Commands
+
+Dotfiles allow users to create custom shortcuts and commands, including aliases, functions and key-bindings, that can tremendously improve productivity. These can be both system level and project specific. In this section, we will look into aliases and functions.
+
+##### Feature 2.1: Aliases
+
+An alias is a shortcut command to a longer and more complex command. A new alias is defined by assigning a string with the command to a name in the format `alias <name>=<command>`. Aliases are often set in the `~/.bashrc` or `~/.zshrc` file. The examples below illustrate the use of aliasing.
+
+```bash
+alias c='clear'
+alias lf='/bin/ls --color -CF'
+alias ll='ls -l --color=auto'
+alias ls='ls --color=auto'
+alias r='fc -s'
+alias vi='vim'
+```
+
+In this case, a user is able to type `c` instead of typing out `clear` in order to clear the terminal screen, a very commonly used command. Aliases are especially useful for abstracting long commands that are used often, and when compounded can save the user a lot of time and effort.
+
+##### Feature 2.2: Functions
+
+Much like aliases, functions allows a user to abstract longer commands into short commands. However, there is additional functionality such as being able to pass command line [arguments](https://tecadmin.net/tutorial/bash-scripting/bash-command-arguments/) and [flags](https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash). This makes the commands more extensible and reuseable for a wide array of functionality.
+
+The example function `say_hello` below takes in 2 arguments, represented by `$1` and `$2`.
+```bash
+say_hello () {
+  echo "hello $1, good $2!"
+}
+```
+If the user were to type the following into the terminal:
+```bash
+say_hello John morning
+```
+it would produce the output:
+```
+hello John, good morning!
+```
```

Comment:
Give an example of a useful function (rather than a toy example), if you can come up with such a function that is also easy to understand?

Reply:
I added 2 new examples here. The second one is more complicated, but it is very useful and I think readers can infer how it works even if they aren't familiar with syntax.